src/Momento.Sdk/Responses/CacheGetBatchResponse.cs

```diff
@@ -1,29 +1,72 @@
 ﻿using System.Collections.Generic;
-using System.Linq;
+using Momento.Sdk.Exceptions;
 
 namespace Momento.Sdk.Responses;
 
 public class CacheGetBatchResponse
 {
-    public List<CacheGetResponse> Responses { get; }
-
-    public CacheGetBatchResponse(IEnumerable<CacheGetResponse> responses)
+    public class Success : CacheGetBatchResponse
     {
-        this.Responses = new(responses);
-    }
+        public List<CacheGetResponse> Responses { get; }
 
-    public IEnumerable<CacheGetStatus> Status
-    {
-        get => Responses.Select(response => response.Status);
-    }
+        public Success(IEnumerable<CacheGetResponse> responses)
+        {
+            this.Responses = new(responses);
+        }
 
-    public IEnumerable<string?> Strings()
-    {
-        return Responses.Select(response => response.String());
+        public IEnumerable<CacheGetStatus> Status
```

Comment:
@malandis you should look at this part.  i'm not super familiar with this part of the API but some random thoughts I have are:

1. We have been getting rid of the enums in other places, and I had assumed that would happen for CacheGet too, but it seems like it's not happening here.
2. do we really need 3 separate methods for this?  would it not be okay to just have a single enumerable that contains the polymorphic parent type?
3. in the loops below it looks like we are assuming that all of the responses are `Success`, but I don't think there is any guarantee that that is actually true?

Reply:
2. You could. I included methods to parallel the unary `Get`: get the status, get the payload. Since this is a batch operation I also included to get the individual items
3. That's correct for the way this is implemented now. If this were a single RPC then there would only be on `Success`